### PR TITLE
Unit tests for legacy privacy group id generation

### PR DESCRIPTION
**Pull request by jframe**

### CODE REVIEW

File: src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java [hunk 1]
```diff
@@ -164,6 +172,34 @@ void encryptGeneratesDifferentCipherForSamePayloadAndKey() {
     assertNotEquals(encryptedPayload1.cipherText(), encryptedPayload2.cipherText());
   }
 
+  @Test
+  public void generatesLegacyPrivacyGroupIdForDuplicateValues() {
+    final String expectedPrivacyGroupId = "/xzRjCLioUBkm5LYuzll61GXyrD5x7bvXzQk/ovJA/4=";
```
Comment (macfarla) [inline src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java]: nit: can the expectedPrivacyGroupId be extracted as a constant to remove dupe

File: src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java [hunk 2]
```diff
@@ -13,22 +13,30 @@
...
+import java.util.Arrays;
 
+import org.apache.tuweni.bytes.Bytes;
 import org.apache.tuweni.crypto.sodium.Box;
+import org.apache.tuweni.crypto.sodium.Box.PublicKey;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 class SodiumEnclaveTest {
+  private static final String PRIVACY_GROUP_ID1 = "negmDcN2P4ODpqn/6WkJ02zT/0w0bjhGpkZ8UP6vARk=";
```
Comment (rain-on) [inline src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java]: nit: I know the code exists in the enclave - but not sure I see these new tests are "enclave" related - not going to die on a hill about it however.

File: src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java (hunk 1 above)
Comment (rain-on) [inline src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java]: nit: add a note for what this value represents? Or is it just magical?

File: src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java [hunk 3]
```diff
@@ -164,6 +172,34 @@ void encryptGeneratesDifferentCipherForSamePayloadAndKey() {
     assertNotEquals(encryptedPayload1.cipherText(), encryptedPayload2.cipherText());
   }
 
+  @Test
+  public void generatesLegacyPrivacyGroupIdForDuplicateValues() {
```
Comment (macfarla) [inline src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java]:
```suggestion
  public void generatesSameLegacyPrivacyGroupIdForDuplicateValues() {
```

File: src/test/java/net/consensys/orion/storage/PrivacyGroupStorageTest.java [hunk 4]
```diff
@@ -128,4 +128,5 @@ void storeTwice() throws InterruptedException {
     final String key2 = payloadStorage.put(privacyGroupPayload2).get();
     assertEquals(key1, key2);
   }
+
```
Comment (rain-on) [inline src/test/java/net/consensys/orion/storage/PrivacyGroupStorageTest.java]: nit: should this newline be here?

Comment (jframe) [reply src/test/java/net/consensys/orion/storage/PrivacyGroupStorageTest.java]: nah not needed. removed

File: src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java (hunk 2 above)
Comment (jframe) [reply src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java]: moved into new test class

File: src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java (hunk 3 above)
Comment (jframe) [reply src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java]: done

File: src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java (hunk 1 above)
Comment (jframe) [reply src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java]: done

Comment (jframe) [reply src/test/java/net/consensys/orion/enclave/sodium/SodiumEnclaveTest.java]: added a note

**Pull request merged**